app/main.py

```diff
@@ -127,6 +121,14 @@ def get_version_number(dataset_version: str) -> int:
     return int(dataset_version[1:])
 
 
+# Uses the dataset_version (i.e. path.stem) to get the version number
```

Comment:
Just to check - is this top comment needed still?

Also, comments might need to live a bit closer to the lines/blocks of code that they're explaining, so could move this inside the `build_dataset_metadata` function near the top, or have it as a docstring maybe?

Reply:
Similar to the above, it is a bit confusing to have a comment that says `this method may not work as intended`

Reply:
Those comments were my own notes and points I wanted to explain and I completely forgot to remove them 😆 they're not needed, I've removed them